common/buildcraft/silicon/tile/TileLaser.java

```diff
@@ -64,50 +71,75 @@ public TileLaser() {
         caps.addProvider(new MjCapabilityHelper(new MjBatteryReceiver(battery)));
     }
 
-    private void findTarget() {
+    @Override
+    public int getUpdateRange() {
+        return TARGETING_RANGE;
+    }
+
+    @Override
+    public BlockPos getSubscriberPos() {
+        return getPos();
+    }
+
+    @Override
+    public void setWorldUpdated(@Nonnull World world, @Nonnull BlockPos eventPos, @Nonnull IBlockState oldState,
+                                @Nonnull IBlockState newState, int flags) {
+        this.worldHasUpdated = true;
+    }
+
+    private void findPossibleTargets() {
+        targetPositions = new ArrayList<>();
         IBlockState state = world.getBlockState(pos);
         if (state.getBlock() != BCSiliconBlocks.laser) {
             return;
         }
         EnumFacing face = state.getValue(BuildCraftProperties.BLOCK_FACING_6);
 
-        List<BlockPos> possible = new ArrayList<>();
-        VolumeUtil.iterateCone(world, pos, face, 6, true, (w, s, p, visible) -> {
+        VolumeUtil.iterateCone(world, pos, face, TARGETING_RANGE, true, (w, s, p, visible) -> {
             if (!visible) {
                 return;
             }
             IBlockState stateAt = world.getBlockState(p);
             if (stateAt.getBlock() instanceof ILaserTargetBlock) {
                 TileEntity tileAt = world.getTileEntity(p);
                 if (tileAt instanceof ILaserTarget) {
-                    ILaserTarget targetAt = (ILaserTarget) tileAt;
-                    if (targetAt.getRequiredLaserPower() > 0) {
-                        possible.add(p);
-                    }
+                    targetPositions.add(p);
+
                 }
             }
         });
+    }
 
-        if (possible.isEmpty()) {
+    private void randomlyChooseTargetPos() {
+        List<BlockPos> targetsNeedingPower = new ArrayList<>();
+        for(BlockPos position: targetPositions) {
+            if (powerNeededAt(position)) {
+                targetsNeedingPower.add(position);
+            }
+        }
+        if (targetsNeedingPower.isEmpty()) {
             targetPos = null;
             return;
         }
-
-        targetPos = possible.get(world.rand.nextInt(possible.size()));
+        targetPos = targetsNeedingPower.get(world.rand.nextInt(targetsNeedingPower.size()));
     }
 
-    private ILaserTarget getTarget() {
-        if (targetPos != null) {
-            TileEntity tile = world.getTileEntity(targetPos);
+    private boolean powerNeededAt(BlockPos position) {
```

Comment:
Perhaps ````isPowerNeededAt````?

Reply:
I like the is but I'm not sure about keeping at. Do you prefer `isPowerNeeded` or `isPowerNeededAt`?

Reply:
I hadn't actually noticed the "at" first time, and just copy-pasted the existing words without thinking. Woops. Seeing as the method takes a blockposition rather than nothing, having the "at" probably makes sense.